lib/active_merchant/billing/gateways/worldpay_online_payments.rb

```diff
@@ -0,0 +1,216 @@
+            "type"=> "Card",
+            "name"=> name,
+            "expiryMonth"=> exp_month,
+            "expiryYear"=> exp_year,
+            "cardNumber"=> number,
+            "cvc"=> cvc
+          },
+          "clientKey"=> @client_key
+        }
+
+        token_response = commit(:post, 'tokens', obj, {'Authorization' => @service_key})
+
+        token_response
+      end
+
+      def create_post_for_auth_or_purchase(token, money, options)
+      {
+        "token" => token,
+        "orderDescription" => options[:description],
+        "amount" => money,
+        "currencyCode" => options[:currency] || default_currency,
+        "name" => options[:billing_address]&&options[:billing_address][:name] ? options[:billing_address][:name] : '',
+        "billingAddress" => {
+          "address1"=>options[:billing_address]&&options[:billing_address][:address1] ? options[:billing_address][:address1] : '',
+          "address2"=>options[:billing_address]&&options[:billing_address][:address2] ? options[:billing_address][:address2] : '',
+          "address3"=>"",
+          "postalCode"=>options[:billing_address]&&options[:billing_address][:zip] ? options[:billing_address][:zip] : '',
+          "city"=>options[:billing_address]&&options[:billing_address][:city] ? options[:billing_address][:city] : '',
+          "state"=>options[:billing_address]&&options[:billing_address][:state] ? options[:billing_address][:state] : '',
+          "countryCode"=>options[:billing_address]&&options[:billing_address][:country] ? options[:billing_address][:country] : ''
+          },
+          "customerOrderCode" => options[:order_id],
+          "orderType" => "ECOM",
+          "authorizeOnly" => options[:authorizeOnly] ? true : false
+        }
+      end
+
+      def parse(body)
+        if (!body)
+          body = {}
+        else
+          body = JSON.parse(body)
+        end
+        body
+      end
+
+      def headers(options = {})
+        headers = {
+          "Authorization" => @service_key,
+          "Content-Type" => 'application/json',
+          "User-Agent" => "Worldpay/v1 ActiveMerchantBindings/#{ActiveMerchant::VERSION}",
+          "X-Worldpay-Client-User-Agent" => user_agent,
+          "X-Worldpay-Client-User-Metadata" => {:ip => options[:ip]}.to_json
+        }
+        if options['Authorization']
+          headers['Authorization'] = options['Authorization']
+        end
+        headers
+      end
+
+      def commit(method, url, parameters=nil, options = {})
+
+        raw_response = response = nil
+        success = false
+        begin
+          if parameters == nil
+            json = nil
+          else
+            json = parameters.to_json
+          end
+
+          raw_response = ssl_request(method, self.live_url + url, json, headers(options))
+
+          if (raw_response != '')
+            response = parse(raw_response)
+            success = !response.key?("httpStatusCode")
+          else
+            success = true
```

Comment:
So, an empty response is deemed a success? That seems a little backwards, but I'm not familiar with this API